src/containers/forms/OrganizationSelect/index.tsx

```diff
@@ -0,0 +1,171 @@
+import reducerRegistry from '@onaio/redux-reducer-registry';
+import React, { useEffect, useState } from 'react';
+import Select from 'react-select';
+import { OpenSRPService } from '../../../services/opensrp';
+
+import { keyBy, values } from 'lodash';
+import { connect } from 'react-redux';
+import { Store } from 'redux';
+import { OPENSRP_ORGANIZATION_ENDPOINT } from '../../../constants';
+import store from '../../../store';
+import assignmentReducer, {
+  Assignment,
+  fetchAssignments,
+  getAssignmentsArrayByPlanIdByJurisdictionId,
+  reducerName as assignmentReducerName,
+} from '../../../store/ducks/opensrp/assignments';
+import organizationsReducer, {
+  fetchOrganizations,
+  getOrganizationsArray,
+  Organization,
+  reducerName as organizationsReducerName,
+} from '../../../store/ducks/opensrp/organizations';
+
+reducerRegistry.register(assignmentReducerName, assignmentReducer);
+reducerRegistry.register(organizationsReducerName, organizationsReducer);
+
+/** react-select Option */
+interface SelectOption {
+  label: string;
+  value: string;
+}
+
+/** OrganizationSelect props */
+export interface OrganizationSelectProps {
+  assignments: Assignment[];
+  fetchAssignmentsAction: typeof fetchAssignments;
+  fetchOrganizationsAction: typeof fetchOrganizations;
+  jurisdictionId: string;
+  name: string; // name of the input
+  organizations: Organization[];
+  planId: string;
+  serviceClass: typeof OpenSRPService /** the OpenSRP service */;
+  values: SelectOption[];
+}
+
+/** default props for OrganizationSelect */
+const defaultProps: OrganizationSelectProps = {
+  assignments: [],
+  fetchAssignmentsAction: fetchAssignments,
+  fetchOrganizationsAction: fetchOrganizations,
+  jurisdictionId: '',
+  name: '',
+  organizations: [],
+  planId: '',
+  serviceClass: OpenSRPService,
+  values: [],
+};
+
+/**
+ * OrganizationSelect - a cascading select for Jurisdictions
+ * Allows you to drill-down Jurisdictions until you select a Focus Area
+ * This is simply a Higher Order Component that wraps around AsyncSelect
+ */
+const OrganizationSelect = (props: OrganizationSelectProps) => {
+  const {
+    fetchAssignmentsAction,
+    fetchOrganizationsAction,
+    jurisdictionId,
+    name,
+    organizations,
+    planId,
+    serviceClass,
+    values: selectOptions,
+  } = props;
+
+  const loadOrganizations = async (service: typeof serviceClass) => {
```

Comment:
We can actually probably drop this as we should already have all the Organizations loaded at this point